cartographer/sensor/map_by_time.h

```diff
@@ -130,7 +132,44 @@ class MapByTime {
     bool operator!=(const ConstIterator& it) const { return !operator==(it); }
 
    private:
-    typename std::map<common::Time, DataType>::const_iterator iterator_;
+    typename TimeIndexedMap::const_iterator iterator_;
+  };
+
+  class ConstTrajectoryIterator {
+   public:
+    using iterator_category = std::bidirectional_iterator_tag;
+    using value_type = int;
+    using difference_type = int64;
+    using pointer = int*;
```

Comment:
Shouldn't this be `const int*`?

Reply:
Do you want to make the change in `MapById` as well?

Reply:
Assuming you do. Updated.

Reply:
Wait, shouldn't it be `int * const`? A pointer to const, not a const pointer?